blockchain/chain.go

```diff
@@ -1302,6 +1302,45 @@ func (b *BlockChain) HeightRange(startHeight, endHeight int32) ([]chainhash.Hash
 	return hashes, nil
 }
 
+// HeightToHashRange returns a range of block hashes for the given start height
+// and end hash, inclusive on both ends.  The hashes are for all blocks that are
+// ancestors of endHash with height greater than or equal to startHeight.  The
+// end hash must belong to a block that is known to be valid.
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) HeightToHashRange(startHeight int32,
+	endHash *chainhash.Hash, maxResults int) ([]chainhash.Hash, error) {
+
+	endNode := b.index.LookupNode(endHash)
+	if endNode == nil {
+		return nil, fmt.Errorf("no known block header with hash %v", endHash)
+	}
+	if !b.index.NodeStatus(endNode).KnownValid() {
```

Comment:
No, it's not on top of the latest changes, but it is built on top of the change that introduced the node statuses into the in-memory representation. The latest change that has not been incorporated persists the validation status for non-main-chain blocks. That doesn't matter for this use case though, because peers should only request cfilters for headers received during the current connection.

Reply:
Gotcha thanks.